Observer/OrderSaveAfter.php

```diff
@@ -57,22 +59,26 @@ function execute(Observer $ob) {
 							? '{"error": "Magento webhook failed to encode order, please look at order ' . $order_id . ' on website to see the details."}'
 							: $encoded_str;
 						$res = $this->post($payload_str, $o->getStore());
+					} catch (OLE $e) {
+						$comment[] = 'Stock2Shop Webhook exception: ' . $e->getMessage();
 					} catch (\Exception $e) {
 						$this->exception_msg = 'Stock2Shop Webhook exception: ' . $e->getMessage();
 						$this->logger->error($this->exception_msg);
```

Comment:
Why doesn't this general Exceptoin handler work as expected? It looks to me like it should log the exception, and add it as a comment.... ? Why the need for special handling of the OrderLineError

And also, why didn't it handle the original error we were getting? I'm a bit confused...

Was that maybe a Throwable - not an Exception?

Reply:
> Why doesn't this general Exceptoin handler work as expected? It looks to me like it should log the exception, and add it as a comment.... ? Why the need for special handling of the OrderLineError

It, more likely than not, does work correctly. But it didn't work because we are catching an `Exception` not an `Error`. 

> Why the need for special handling of the OrderLineError

I don't think the `$this->logger->error($this->exception_msg);`  adds the respective message to the comments. The reason why I added a separate `OrderLineError`, was to append the respective details to the comment which will be displayed as a note on the order. (Question) I don't think the `$this->logger->error()` adds the message as a note on the order 🤔

> Was that maybe a Throwable - not an Exception?

The original error we were receiving was a `TypeError`.

Reply:
[99b18d5](https://github.com/stock2shop/magento2_module_webhook/pull/12/commits/99b18d53e000c5cd36a8f8acd195546dda8ea2bd)

[fc29aab](https://github.com/stock2shop/magento2_module_webhook/pull/12/commits/fc29aab9979688acfdba83c1b29a3f46e690998e)